### PR TITLE
Bjj 8 reinitialize pokemons

### LINKED ISSUE #8: FS 1 db reinitialize pokemons
### Feature
Right now our server is just acting as a middle man, and still making a call to pokeapi... this should not be the case .. now that we know everything that want on a pokemon document in the bd .. we can name copy exactly what we want on our own db and start allowing the client to pull from the db that we manage instead of a 3rd party.

### Task
1. clear db of pokemon to allow for new
2. map the incoming api data for fit that of the new document
3. save said data to our own db
4. ref index.js to make the request to a local db instead of pokeapi

---
#### Additional Notes
I will be sending another pokemons.json file to our chat that will need to be added in place of the one that exists on each on yall local system's mongo db

### PR DESCRIPTION
**Pull request by brycejjacks**
### Purpose
in out current server setup, we have been working as the middle man, now that we have finalized out data, we need to store the list of pokemon in mongo db and request our info from there

### Changes
1. reinit pokemons
2. pull pokemon list from db
3. note seed data so it is not lost in oblivion

#### Additional Notes
I will be sending a new pokemons.json .. be on the lookout

Closes #8

### CODE REVIEW

File: src/server/index.js [hunk 1]
```diff
@@ -27,25 +27,11 @@ app.use(cors());
...
-            // UNCOMMENT NEXT LINE FOR SEED DATA 
-            // new PokemonModel(pokemon).save().catch(e => console.log(e))
-            return pokemon
-        })
-        res.json(pokemons)
-    });
+    PokemonModel.find().lean()
+    .then(pokemons => res.json(pokemons))
+    .catch(e => {
+        res.sendStatus(500).json(e)
```
Comment (IvanPiceno171) [inline src/server/index.js:33]: what is sendStatus doing?

File: src/server/index.js [hunk 2]
```diff
@@ -27,25 +27,11 @@ app.use(cors());
...
-                type: result.types.map((type) => type.type.name).join(', '), // if remove .join() then type = []
-                id: result.id
-            }
-            // UNCOMMENT NEXT LINE FOR SEED DATA 
-            // new PokemonModel(pokemon).save().catch(e => console.log(e))
-            return pokemon
-        })
-        res.json(pokemons)
-    });
+    PokemonModel.find().lean()
```
Comment (IvanPiceno171) [inline src/server/index.js:30]: what does  .lean() method do?

**Review by IvanPiceno171 — APPROVED** [review]
Ok merge after resolving conversations

File: src/server/index.js (hunk 1 above)
Comment (brycejjacks) [reply src/server/index.js:33]: send status give a brief idea about how the server was able to handle the request. 404 says that the endpoint is not found, 403 means forbidden, 500 means internal server error.. there are other codes to look up

File: src/server/index.js (hunk 2 above)
Comment (brycejjacks) [reply src/server/index.js:30]: model.find returns a large mongo document.. adding lean to the chain cuz away unneeded data and simplifies it just the obj at hand.. this code will still work the same way without .lean() maybe just a tad bit slower

**Pull request merged**